packages/core/src/request.ts

```diff
@@ -58,11 +58,50 @@ export function eventToSentryRequest(event: Event, api: APIDetails): SentryReque
   const { transactionSampling } = event.sdkProcessingMetadata || {};
   const { method: samplingMethod, rate: sampleRate } = transactionSampling || {};
 
+  // This is part of the hack below - in theory, all events should be marked as having gone through normalization and so
+  // we should never set this tag
+  if (!(event.sdkProcessingMetadata && event.sdkProcessingMetadata.baseClientNormalized)) {
+    event.tags = { ...event.tags, skippedNormalization: true };
+  }
+
   // prevent this data from being sent to sentry
   delete event.sdkProcessingMetadata;
 
+  // This is a temporary hack in order to debug a serialization error - see
+  // https://github.com/getsentry/sentry-javascript/issues/2809 and
+  // https://github.com/getsentry/sentry-javascript/pull/4425. TL;DR: even though we normalize all events (which should
+  // prevent this), something is causing `JSON.stringify` to throw a circular reference error.
+  //
+  // TODO: It was necessary to compensate for this in the browser playwright tests, so when it's time to pull this hack
+  // out, search for `skippedNormalization` and you'll find a note pointing out what to remove.
+  const withSDKInfo = sdkInfo ? enhanceEventWithSdkInfo(event, api.metadata.sdk) : event;
+  let body;
+  try {
+    // 99.9% of events should get through just fine - no change in behavior for them
+    body = JSON.stringify(withSDKInfo);
+  } catch (err) {
+    // Record data about the error without replacing original event data, then force renormalization
+    event.tags = { ...event.tags, JSONStringifyError: true };
+    event.extra = { ...event.extra, JSONStringifyError: err };
```

Comment:
```suggestion
    withSDKInfo.tags = { ...withSDKInfo.tags, JSONStringifyError: true };
    withSDKInfo.extra = { ...withSDKInfo.extra, JSONStringifyError: err };
```

Reply:
Or rename it to something like `enhancedEvent` to be more generic.

Reply:
> Or rename it to something like `enhancedEvent` to be more generic.

Heh. Just realized the function mutates the event in place. No variable needed!

Reply:
> tbh I'd rather we must mutate the object directly, saves on bundle size:

I switched to object mutation (though it did require me to add 

```
  event.tags = event.tags || {};
  event.extra = event.extra || {};
  ```

so I'm honestly not sure it's any shorter in the end), but I didn't bother with the constant because this is hopefully coming right back out I a week or two.